src/services/event.ts

```diff
@@ -286,6 +298,42 @@ export async function findEventsByRegionId(id: number): Promise<EventShape[] | n
   return findEventHelper({ regionId: id }, true) as Promise<EventShape[]>;
 }
 
+export async function findEventCreators(regionId: number): Promise<EventShape[] | null> {
```

Comment:
I know we talked about this, but I really don't think you need this handler - the handler in users/services.js that gets stored on the frontend as "additionalData" has all the users this retrieves as "collaborators" already, this is literally the exact same query

Reply:
adding the owner would be a one-liner on the front-end

Reply:
The handler for this handles a use case where the IST (owner) might not come back with the available creators.

Do we still want to change this?

Reply:
```js 
const currentOwner = creators.find((creator) => creator.id === ownerId);

if (!currentOwner) {
     // If the current ownerId is not in the creators array, add it.
    const owner = await userById(ownerId);
    creators.push({ id: owner.id, name: owner.name });
 }
 ```
 
 could be ported to the frontend with one less DB query, right? We already have the user on the frontend so assuming we access the UserContext we'd just check to see if they were an admin, and run this exact logic on the options for the select, something like:
 
 ```js
  const { user } = useContext(UserContext);
  const hasAdminRights = isAdmin(user);
  const { users: { collaborators, pointOfContact, eventCreators } } = additionalData;

  const ownerName = owner && owner.name ? owner.name : '';

  const eventCreatorOptions = !eventCreators
    ? []
    : eventCreators.map((o) => ({ value: o.id, label: o.name }));
 
 const currentOwner = eventCreatorOptions.find((creator) => user.id === creator.value);
 if (!currentOwner) {
     // If the current ownerId is not in the creators array, add it.
    eventCreatorOptions.push({ id: user.id, name: user.name });
 }
 
 
 ```

Reply:
2 less DB queries, since we don't have to get userbyid again either